drivers/tee/optee/optee.c

```diff
@@ -177,9 +178,189 @@ static void u64_to_regs(uint64_t val, uint32_t *reg0, uint32_t *reg1)
 	*reg1 = val;
 }
 
-static void handle_rpc_call(const struct device *dev, struct optee_rpc_param *param)
+static inline bool check_param_input(struct optee_msg_arg *arg)
+{
+	return !(arg->num_params != 1 ||
+		 arg->params[0].attr != OPTEE_MSG_ATTR_TYPE_VALUE_INPUT);
```

Comment:
Oh well... Any particular reason why you can't rewrite it as 
```
return arg->num_params == 1 &
		 arg->params[0].attr == OPTEE_MSG_ATTR_TYPE_VALUE_INPUT
```
? 

It is really hard to read and do all logical conversions in my head.

This is the security-related code in a safety-oriented OS. Code must be as simple and as dumb as possible

Reply:
fixed